plugins/modules/pnp_intent.py

```diff
@@ -1008,36 +600,95 @@ def get_diff_merged(self):
             class instance for further use.
         """
 
-        if not self.have.get("device_found"):
-            self.log("Adding device to pnp database")
-            response = self.dnac_apply['exec'](
-                family="device_onboarding_pnp",
-                function="add_device",
-                params=self.want.get("pnp_params"),
-                op_modifies=True,
-            )
-
-            self.have["device_id"] = response.get("id")
-            self.log(str(response))
-            self.log(self.have.get("device_id"))
+        device_count_params = {
+            "serial_number": self.want.get("serial_number"),
+            "state": "Provisioned"
+        }
 
-        claim_params = self.get_claim_params()
-        claim_response = self.dnac_apply['exec'](
-            family="device_onboarding_pnp",
-            function='claim_a_device_to_a_site',
-            op_modifies=True,
-            params=claim_params,
-        )
+        if not self.have.get("device_found"):
+            if not self.want["add_device_method"]:
+                self.module.fail_json(msg="Device needs to be added before claiming", response=[])
+            else:
+                if not self.want["site_name"]:
+
+                    if self.want["add_device_method"] == "Single":
+                        self.log("Adding device to pnp database")
+                        dev_add_response = self.dnac_apply['exec'](
+                            family="device_onboarding_pnp",
+                            function="add_device",
+                            params=self.want.get("pnp_params"),
+                            op_modifies=True,
+                        )
+
+                        self.have["deviceInfo"] = dev_add_response.get("deviceInfo")
+                        self.log(str(dev_add_response))
+                        if self.have["deviceInfo"]:
+                            self.result['msg'] = "Only Device Added Successfully"
+                            self.result['response'] = dev_add_response
+                            self.result['diff'] = self.validated_config
+                            self.result['changed'] = True
+                        else:
+                            self.module.fail_json(msg="Device Addition Failed", response=[])
```

Comment:
set and return instead of failing Json. this should be taken care in check_return_status.... We can check the failure condition and return from there itself.